### DISCUSSION TITLE
Contact algorithm

### DISCUSSION THREAD

**Discussion by QZheng82** [Anura3D/Anura3D_OpenSource#100] [Q&A] [answered]
May I ask if I should not use the contact algorithm if the deformation of two objects in contact is too large?Instead, should we use a moving mesh?

**Comment by ayerrocolom** [comment, marked as answer]
Hello!
the contact algorithm implemented in Anura3D should only be used if one of the bodies in contact is defined always by the same nodes. This means that we have two options:
(1) One of the bodies is not expected to deform (e.g., rigid or with a large stiffness) and fixed in space (e.g. check chapter 5 in the tutorial manual)
(2) One of the bodies is not expected to deform (e.g., rigid or with a large stiffness) and moves with a linear trajectory (either vertically or horizontally, following the reference coordinate system of the problem). In this case, you can use a moving mesh technique that follows such body maintaining the definition of the contact at the same set of nodes (e.g., check chapter 8 in the tutorial manual)

**Reply by JayfansChou** [reply]
Hi.
May I ask a question about the contact problem in GiD-Anura3D? Please let me take the impact problem in Tutorial Manual 2023, if the impact part is defined as a rigid body, then is it necessary to define the moving mesh (I mean the whole component including compressing, moving mesh, and mesh reference material)? And how to understand the meaning of the content described in '9.3.9 Defining the moving mesh' of Tutorial Manual 2023, which says as follows:
![Notes](https://github.com/user-attachments/assets/8d9db75d-b035-492d-837a-6019e841e64b)
I'd appreciate it for your help.

**Reply by elbarrendero** [reply]
i get the following warning...
 [WARNING] : (time step:      1) Body contact can not be assigned to more than one material

I'm trying to capture contact between 3 surfaces.. selecting different elements for each contact surface properties (2 contacts surfaces).. is this correct? 
<img width="2560" height="1540" alt="image" src="https://github.com/user-attachments/assets/ca5bcb08-e95a-4ed4-8427-6c2b4e185a52" />

**Reply by icy1237** [reply]
> 我收到以下警告...... [警告]：（时间步长：1）身体接触不能分配给多个材料
> 
> 我正在尝试捕捉三个表面之间的接触......为每个接触面属性选择不同的元素（两个接触面）......这样说对吗？ <img alt="图片" width="2000" height="1540" src="https://private-user-images.githubusercontent.com/45673642/558722874-ca5bcb08-e95a-4ed4-8427-6c2b4e185a52.png?jwt=eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NzMwMTg4NzcsIm5iZiI6MTc3MzAxODU3NywicGF0aCI6Ii80NTY3MzY0Mi81NTg3MjI4NzQtY2E1YmNiMDgtZTk1YS00ZWQ0LTg0MjctNmMyYjRlMTg1YTUyLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNjAzMDklMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjYwMzA5VDAxMDkzN1omWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTJhNWRkZWYyNzgwMDgwNDM0ODVhMTRiZWRjZWU1MTJkYWU4MjE1MDI3YmJhYTVhNzY0ZWRiZDkxNDRmOTg0NDMmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.rVCi1Yj1_hGhpcwC6-onM0pn6uixBkTQgDzIaQaSyUA">

Hello!

I've encountered the same problem. Have you solved it?

**Reply by mikael-onsilva** [reply]
Follow Tutorial 6 from the manual. The contact algorithm must be applied only to the foundation, while the MATERIAL 1, 2 or 3 refers to the materials in contact with the base (undeformed). This means you can't apply the algorithm to both tailings-dam and tailings-foundation contact. As stated before by @ayerrocolom, the base layer, in your case, must be the foundation, as the most rigid.

By the end you can have MATERIAL 1 as you tailings layer and MATERIAL 2 as  the dam, with the contact applied to the foundation.
